biostudiesclient/models.py

```diff
@@ -0,0 +1,87 @@
+"""
+biostudiesclient.models
+~~~~~~~~~~~~
+
+This module contains the data models for BioStudy entities.
+
+:copyright: (c) 2021 by Karoly Erdos.
+:license: Apache2, see LICENSE for more details.
+"""
+
+import dataclasses
+import json
+from dataclasses import dataclass
+from typing import Set
+
+
+@dataclass
+class BioStudiesAttribute:
+    """ Data class with name, value pairs; that represent a generic BioStudy attribute. """
+    name: str = None
+    value: str = None
+
+    def __hash__(self):
+        return hash((self.name, self.value))
+
+
+@dataclass
+class BioStudyFile:
+    """ Data class that represent a file for BioStudies archive. """
+    path: str = None
+    file_type: str = None
+    attributes: Set[BioStudiesAttribute] = None
```

Comment:
Can we have a constructor for these `@dataclass` classes that initialises the sets to be an empty set?
Or if having a constructor is a problem, default them to the empty set some other way.

This would effect all the `.attributes` `.files` and `.links` locals below.

Reply:
I added default factory method to initialise these sets. Thanks